### PR TITLE
chore(infra/biome): rule noUselessElse

### PR DESCRIPTION
**Pull request by shulaoda**
## Summary
Related to issue #7182 

## Checklist

- [ ] Tests updated (or not required).
- [ ] Documentation updated (or not required).

### CODE REVIEW

File: biome.jsonc [hunk 1]
```diff
@@ -49,12 +49,14 @@
...
 				"useExportType": "error",
-				"useEnumInitializers": "error"
+				"useEnumInitializers": "error",
+				"noUselessElse": "error"
 			}
 		},
 		"ignore": [
 			"tests/**/*",
 			"crates/**/*",
+			"website/doc_build",
```
[the remark this answers is not in the document]
Comment (shulaoda) [reply biome.jsonc:59]: Yes, I found that it scans files in `docs_build`, you can try it out.

[the remark this answers is not in the document]
Comment (shulaoda) [reply biome.jsonc:59]: > we already have enabled `"vcs"` Is it necessary to ignore`doc_build`?

Yes, I found that it scans files in docs_build, you can try it out.

[the remark this answers is not in the document]
Comment (SoonIter) [reply biome.jsonc:63]: I'll check it later and investigate whether there is a bug of biome.
This is a big mr, maybe it needs to be rebased and resolved conflicts several times. :D
Let me merge other small mrs

[the remark this answers is not in the document]
Comment (SoonIter) [reply biome.jsonc:63]: I investigate this issue
biome only check the ignore file in `biome.jsonc` directory, and our `.gitignore` file under `website/.gitignore` is ignored

we can add `doc_build` to `/.gitignore` temporarily and open a issue in biome

[the remark this answers is not in the document]
Comment (SoonIter) [reply biome.jsonc:63]: https://github.com/biomejs/biome/issues/2312

**Review by SoonIter — APPROVED** [review]

**Pull request merged**